src/feedretriever.cpp

```diff
@@ -0,0 +1,304 @@
+#include "feedretriever.h"
+
+#include <cinttypes>
+
+#include "cache.h"
+#include "config.h"
+#include "configcontainer.h"
+#include "curlhandle.h"
+#include "freshrssapi.h"
+#include "logger.h"
+#include "minifluxapi.h"
+#include "newsblurapi.h"
+#include "ocnewsapi.h"
+#include "remoteapi.h"
+#include "rss/parser.h"
+#include "rssignores.h"
+#include "strprintf.h"
+#include "ttrssapi.h"
+#include "utils.h"
+
+namespace newsboat {
+
+FeedRetriever::FeedRetriever(ConfigContainer& cfg, Cache& ch, RssIgnores* ign,
+	RemoteApi* api, CurlHandle* easyhandle)
+	: cfg(cfg)
+	, ch(ch)
+	, ign(ign)
+	, api(api)
+	, easyhandle(easyhandle)
+{
+}
+
+rsspp::Feed FeedRetriever::retrieve(const std::string& uri)
+{
+	bool is_ttrss = cfg.get_configvalue("urls-source") == "ttrss";
+	bool is_newsblur = cfg.get_configvalue("urls-source") == "newsblur";
+	bool is_ocnews = cfg.get_configvalue("urls-source") == "ocnews";
+	bool is_miniflux = cfg.get_configvalue("urls-source") == "miniflux";
+	bool is_freshrss = cfg.get_configvalue("urls-source") == "freshrss";
```

Comment:
Nit: can these be turned into local variables while we're at it? Maybe even inline them into `if`s, because each of them is only used in one condition.

Reply:
I think this comment was made in one of the intermediate commits.
In the full PR, they were already made into local variables.

However, moving them inline is still a nice improvement, I will do that in a moment.